### PR TITLE
M56D improvements

### PR DESCRIPTION
**Pull request by JeserTheFox**
:cl: jeser
tweak: Synths can no longer fire the M56D.
tweak: The M56D can only be manned by standing on its back.
tweak: It's no longer needed to screwdrive the mount to the ground.
tweak: Burst-fire mode switch is now done through the toggle-mg-burst-fire verb in Weapons tab. Ctrl-clicking will make you shoot in burst-fire if you are in single-fire mode, and vice-versa.
fix: Fixed synth-gun-use-related message bug.
fix: Fixed sprite bug after installing gun that has ammo on a tripod.
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
Conflict.

File: code/modules/cm_marines/smartgun_mount.dm [hunk 1]
```diff
@@ -557,41 +539,59 @@
 
 /obj/machinery/m56d_hmg/on_set_interaction(mob/user)
 	flags_atom |= RELAY_CLICK
-	if(zoom)
-		user.client.change_view(12)
+	user.client.change_view(view_tiles)
+	switch(src.dir)
```
Comment (Rohesie) [inline code/modules/cm_marines/smartgun_mount.dm]:
```suggestion
	switch(dir)
```

File: code/modules/cm_marines/smartgun_mount.dm [hunk 2]
```diff
@@ -58,6 +58,8 @@
 
 /obj/item/device/m56d_gun/examine(mob/user as mob) //Let us see how much ammo we got in this thing.
 	..()
```
Comment (Rohesie) [inline code/modules/cm_marines/smartgun_mount.dm]:
```suggestion
	. = ..()
```

File: code/modules/cm_marines/smartgun_mount.dm [hunk 3]
```diff
@@ -220,55 +206,29 @@
...
+	density = TRUE
 	layer = ABOVE_MOB_LAYER //no hiding the hmg beind corpse
 	use_power = 0
 	var/rounds = 0 //Have it be empty upon spawn.
 	var/rounds_max = 700
 	var/fire_delay = 4 //Gotta have rounds down quick.
 	var/last_fired = 0
-	var/burst_fire = 0 //0 is non-burst mode, 1 is burst.
-	var/safety = 0 //Weapon safety, 0 is weapons hot, 1 is safe.
+	var/burst_fire = FALSE//0 is non-burst mode, 1 is burst.
```
Comment (Rohesie) [inline code/modules/cm_marines/smartgun_mount.dm]:
```suggestion
	var/burst_fire = FALSE
```

File: code/modules/cm_marines/smartgun_mount.dm [hunk 4]
```diff
@@ -220,55 +206,29 @@
...
+	var/burst_fire_toggled = FALSE
+	var/safety = FALSE //Weapon safety, 0 is weapons hot, 1 is safe.
```
Comment (Rohesie) [inline code/modules/cm_marines/smartgun_mount.dm]:
```suggestion
	var/safety = FALSE
```
No need for these explanations if we don't use magic numbers.

File: code/modules/cm_marines/smartgun_mount.dm [hunk 5]
```diff
@@ -399,7 +353,8 @@
 	update_health(rand(M.xeno_caste.melee_damage_lower,M.xeno_caste.melee_damage_upper))
 
 /obj/machinery/m56d_hmg/proc/load_into_chamber()
-	if(in_chamber) return 1 //Already set!
+	if(in_chamber)
+		return 1 //Already set!
```
Comment (Rohesie) [inline code/modules/cm_marines/smartgun_mount.dm]:
```suggestion
		return TRUE //Already set!
```

File: code/modules/cm_marines/smartgun_mount.dm [hunk 6]
```diff
@@ -399,7 +353,8 @@
...
 	if(rounds == 0)
 		update_icon() //make sure the user can see the lack of ammo.
 		return 0 //Out of ammo.
```
Comment (Rohesie) [inline code/modules/cm_marines/smartgun_mount.dm]:
```suggestion
		return FALSE //Out of ammo.
```

**Comment by Rohesie** [comment]
Stylistic changes, this is ready for testing now, and for merging once they are fixed.

File: code/modules/cm_marines/smartgun_mount.dm [hunk 7]
```diff
@@ -128,15 +132,12 @@
 
 /obj/machinery/m56d_post/examine(mob/user)
 	..()
-	if(!anchored)
-		to_chat(user, "It must be <B>screwed</b> to the floor.")
-	else if(!gun_mounted)
+	if(!gun_mounted)
 		to_chat(user, "The <b>M56D Mounted Smartgun</b> is not yet mounted.")
```
Comment (DominikPanic) [inline code/modules/cm_marines/smartgun_mount.dm]: `to_chat(user, "The <b>M56D Mounted Smartgun</b> is not yet mounted.")`

`to_chat(user, "The <b>[src]</b> is not yet mounted.")`

**Pull request merged**